### ISSUE TITLE
php-nginx-dev:7.4-alpine / xDebug 3.0.1 and phpStorm

### ISSUE THREAD

**Issue by ekergreis** [webdevops/Dockerfile#383]
Hi,

I use phpStorm 2020.3 and the container "webdevops/php-nginx-dev:7.4-alpine" with new xDebug 3.0.1
No trigger is detected on phpStorm although I pass XDEBUG_SESSION to the request.
I was able to debug on phpStorm with the old version of xdebug.

Could you help me ?
Thanks.

![docker-compose](https://user-images.githubusercontent.com/57322037/102016648-46de5f80-3d62-11eb-86e9-7f03cf0f8671.png)
![phpStorm_1](https://user-images.githubusercontent.com/57322037/102016432-46919480-3d61-11eb-9458-b23dce7aa6b2.PNG)
![phpStorm_2](https://user-images.githubusercontent.com/57322037/102016433-46919480-3d61-11eb-916f-1a2b9a778944.PNG)

**Comment by hhoechtl** [comment]
Xdebug changed the port. You need to reconfigure phpstorm to the new port
9003 https://xdebug.org/docs/upgrade_guide

Emmanuel Kergreis <notifications@github.com> schrieb am So., 13. Dez. 2020,
16:45:

**Comment by ekergreis** [comment]
Thanks but even with the port 9003 in phpStorm it doesn't work.
phpStorm version 2020.3 include a parameter "_Additionnally listen on Xdebug 3 default port '9003'_ "
Additional information, I am under windows with WSL. Everything works except xDebug.

![2020-12-13_19h09_59](https://user-images.githubusercontent.com/57322037/102020017-de4dad80-3d76-11eb-927c-abc75245e6ae.png)

**Comment by hhoechtl** [comment]
I did a test with the following:

```yaml
version: '3.7'

services:
  app:
    image: webdevops/php-nginx-dev:8.0
    volumes:
      - ./:/app
    ports:
      - 8888:80
    environment:
      XDEBUG_CLIENT_HOST: 172.17.0.1
```

The env variable `XDEBUG_CLIENT_HOST` defines the IP adress of my host machines docker bridge interface, as on linux the IP address visiting the container is `127.0.0.1` and therefore the automatic `remote_connect_back` of XDEBUG doesn't work out of the box.
This is probably the same on WSL.

**Comment by ekergreis** [comment]
Thanks, I have tested with 

```
version: '2'
services:
  permanences:
    image: 'webdevops/php-nginx-dev:7.4-alpine'
    working_dir: /var/www/html
    environment:
      - WEB_DOCUMENT_ROOT=/var/www/html/public
      - PHP_DISPLAY_ERRORS=1
      - PHP_MEMORY_LIMIT=2048M
      - PHP_MAX_EXECUTION_TIME=-1
      - XDEBUG_IDE_KEY=PHPSTORM
      - XDEBUG_CLIENT_HOST=172.19.0.1
    volumes:
      - '/home/eloman/http/permanences:/var/www/html:rw,cached'
    ports:
      - '80:80'
```

**phpStorm validator debugger configurator seem to be OK with the config but nothing activate a debug session.
In the container webdevops/php-nginx-dev:7.4-alpine, the 9003 port is not open.**

![2020-12-14_20h45_37](https://user-images.githubusercontent.com/57322037/102127944-6ef6bd00-3e4d-11eb-88c7-118957344ff3.png)

![2020-12-14_20h37_47](https://user-images.githubusercontent.com/57322037/102127032-402c1700-3e4c-11eb-9b42-37d5e9e897c4.png)

**Comment by hhoechtl** [comment]
I think you misunderstand how XDebug works. The port is not open inside the container and you're not connecting to the container. The PHP inside the container is connecting to your PHPStorm which has the ports 9000 and 9003 open to receive incoming connections.
The term "client" is a bit misleading in the XDebug world as the server listening for incoming connections is your PHPStorm and you tell the PHP how to connect to it via `XDEBUG_CLIENT_HOST`.

It does try to connect to PHPStorm if you trigger it either via cookie or GET parameter as described here: http://xdebug.org/docs/step_debug

**Comment by ekergreis** [comment]
Thanks I understand better xDebug works.
![xdebug](https://user-images.githubusercontent.com/57322037/102271014-08da6a80-3f1f-11eb-855b-d2215c5679fc.gif)

I understand my problem is certainly related to WSL and I will try to check the correct settings.
When I trigger request to debug (with the XDEBUG_SESSION_START parameter), I identified the following error in the container logs : _Xdebug: [Step Debug] Could not connect to debugging client. Tried: 172.19.0.1:9003 (from REMOTE_ADDR HTTP header), 172.19.0.1:9003 (fallback through xdebug.client_host/xdebug.client_port)_

![log_request](https://user-images.githubusercontent.com/57322037/102271307-8605df80-3f1f-11eb-93ce-6e0d0b4f4917.png)

I will keep you informed if I find a solution.
Thanks a lot for your help.
